### PR TITLE
[SMALLFIX] Adding journal replay tests for TTL.

### PR DESCRIPTION
**Pull request by jsimsa**

### CODE REVIEW

File: core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java [hunk 1]
```diff
@@ -713,16 +690,36 @@ public void loadMetadata() throws Exception {
...
+    // Simulate restart.
+    stopServices();
+    startServices();
+
+    FileInfo fileInfo = mFileSystemMaster.getFileInfo(fileId);
+    Assert.assertEquals(fileInfo.getFileId(), fileId);
     HeartbeatScheduler.execute(HeartbeatContext.MASTER_TTL_CHECK);
     mThrown.expect(FileDoesNotExistException.class);
     mFileSystemMaster.getFileInfo(fileId);
+
```
Comment (apc999) [inline core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java]: unnecessary emptyline

File: core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java [hunk 2]
```diff
@@ -731,13 +728,134 @@ public void createFileWithTtl() throws Exception {
    * set in the {@link CreateDirectoryOptions} after the TTL check was done once.
    */
   @Test
-  public void createDirectoryWithTtl() throws Exception {
+  public void ttlDirectoryDelete() throws Exception {
     CreateDirectoryOptions directoryOptions =
         CreateDirectoryOptions.defaults().setRecursive(true).setTtl(0);
-    mFileSystemMaster.createDirectory(NESTED_DIR_URI, directoryOptions);
+    long fileId = mFileSystemMaster.createDirectory(NESTED_DIR_URI, directoryOptions);
```
Comment (apc999) [inline core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java]: NI: inodeId or dirId? up to you

File: core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java [hunk 3]
```diff
@@ -731,13 +728,134 @@ public void createFileWithTtl() throws Exception {
...
+  }
+
+  /**
+   * Tests that TTL delete of a directory is not forgotten across restarts.
+   */
+  @Test
+  public void ttlDirectoryDeleteReplay() throws Exception {
+    CreateDirectoryOptions directoryOptions =
+        CreateDirectoryOptions.defaults().setRecursive(true).setTtl(0);
+    long fileId = mFileSystemMaster.createDirectory(NESTED_DIR_URI, directoryOptions);
```
Comment (apc999) [inline core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java]: same here, dirId or inodeId?

File: core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java [hunk 4]
```diff
@@ -731,13 +728,134 @@ public void createFileWithTtl() throws Exception {
...
+    // Simulate restart.
+    stopServices();
+    startServices();
+
+    FileInfo fileInfo = mFileSystemMaster.getFileInfo(fileId);
+    Assert.assertEquals(fileInfo.getFileId(), fileId);
+    HeartbeatScheduler.execute(HeartbeatContext.MASTER_TTL_CHECK);
+    mThrown.expect(FileDoesNotExistException.class);
+    mFileSystemMaster.getFileInfo(fileId);
+
```
Comment (apc999) [inline core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java]: unnecessary empty line

File: core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java (hunk 1 above)
Comment (jsimsa) [inline core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java]: done

File: core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java (hunk 2 above)
Comment (jsimsa) [inline core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java]: done

File: core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java (hunk 3 above)
Comment (jsimsa) [inline core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java]: done

File: core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java (hunk 4 above)
Comment (jsimsa) [inline core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java]: done

**Pull request merged**